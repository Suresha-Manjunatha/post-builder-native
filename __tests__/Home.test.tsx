/** @format */

import { render } from "@testing-library/react-native";
import React from "react";
import Home from "../src/screens/home";

jest.mock("@react-navigation/native", () => {
	return {
		useNavigation: () => ({
			navigate: jest.fn(),
		}),
	};
});

describe("Home", () => {
	jest.useFakeTimers();

	it("renders without crashing", () => {
		render(<Home />);
	});

	it("matches snapshot", () => {
		const tree = render(<Home />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("has 1 child", () => {
		const tree = render(<Home />).toJSON();
		//@ts-ignore
		expect(tree?.children?.length).toBe(1);
	});

	it("has component with req testID", () => {
		const tree = render(<Home />);
		//@ts-ignore
		expect(tree.getByTestId("home")).toBeTruthy();
	});
});
