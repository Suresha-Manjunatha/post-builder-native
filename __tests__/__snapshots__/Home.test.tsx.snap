// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "padding": 10,
    }
  }
  testID="home"
>
  <RCTScrollView
    ListFooterComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="home-list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
